File: frontend/src/types/backup.ts
export interface Backup {
  id: string
  user_id: string
  profile_id: string
  file_name: string
  file_size: number
  checksum_sha256: string
  backup_type: string
  status: string
  created_at: string
}

export interface BackupCreatePayload {
  profile_id: string
  backup_name: string
  backup_type: string
}

export interface BackupRestorePayload {
  validate_only: boolean
  clear_cache_flag: boolean
  restore_config_only: boolean
}

export interface BackupRestoreResponse {
  success: boolean
  message: string
  checksum_verified: boolean
  restored_profile_id?: string
}

File: backend/app/schemas/backup.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, Field


class BackupCreateRequest(BaseModel):
    profile_id: UUID
    backup_name: str = Field(min_length=1, max_length=100)
    backup_type: str = "CONFIG_ONLY"


class BackupResponse(BaseModel):
    id: UUID
    user_id: UUID
    profile_id: UUID
    file_name: str
    file_size: int
    checksum_sha256: str
    backup_type: str
    status: str
    created_at: datetime

    class Config:
        from_attributes = True


class BackupRestoreRequest(BaseModel):
    validate_only: bool = False
    clear_cache_flag: bool = True
    restore_config_only: bool = True


class BackupRestoreResponse(BaseModel):
    success: bool
    message: str
    checksum_verified: bool
    restored_profile_id: Optional[UUID] = None